diff -ruN libg3d-0.0.3.orig/src/Makefile.am libg3d-0.0.3/src/Makefile.am
--- libg3d-0.0.3.orig/src/Makefile.am	2008-01-11 21:21:55.000000000 +0000
+++ libg3d-0.0.3/src/Makefile.am	2008-01-11 21:22:23.000000000 +0000
@@ -26,4 +26,5 @@
 	-version-number "$(subst .,:,@VERSION@)"
 
 libg3d_la_LIBADD = \
-	@GLIB_LIBS@
+	@GLIB_LIBS@ \
+	-lm
diff -ruN libg3d-0.0.3.orig/src/Makefile.in libg3d-0.0.3/src/Makefile.in
--- libg3d-0.0.3.orig/src/Makefile.in	2008-01-11 21:21:55.000000000 +0000
+++ libg3d-0.0.3/src/Makefile.in	2008-01-11 21:22:14.000000000 +0000
@@ -183,7 +183,8 @@
 
 
 libg3d_la_LIBADD = \
-	@GLIB_LIBS@
+	@GLIB_LIBS@ \
+	-lm
 
 subdir = src
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
